pkg/dataobj/predicate.go

```diff
@@ -1,13 +1,16 @@
 package dataobj
 
 import (
+	"fmt"
+	"strings"
 	"time"
 )
 
 type (
 	// Predicate is an expression used to filter entries in a data object.
 	Predicate interface {
 		isPredicate()
+		String() string
```

Comment:
general question: do we want to make `String() string` a required method on every Predicate? Or may some of them just happen to implement String?